cephfs/admin/clone_progress_test.go

```diff
@@ -0,0 +1,73 @@
+//go:build main
+
+package admin
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCloneProgress(t *testing.T) {
+	fsa := getFSAdmin(t)
+	volume := "cephfs"
+	group := "Park"
+	subname := "Jurrasic"
+	snapname := "dinodna1"
+	clonename := "babydino"
+
+	err := fsa.CreateSubVolumeGroup(volume, group, nil)
+	assert.NoError(t, err)
+	defer func() {
+		err := fsa.RemoveSubVolumeGroup(volume, group)
+		assert.NoError(t, err)
+	}()
+
+	svopts := &SubVolumeOptions{
+		Mode: 0750,
+		Size: 20 * gibiByte,
+	}
+	err = fsa.CreateSubVolume(volume, group, subname, svopts)
+	assert.NoError(t, err)
+	defer func() {
+		err := fsa.RemoveSubVolume(volume, group, subname)
+		assert.NoError(t, err)
+	}()
+
+	err = fsa.CreateSubVolumeSnapshot(volume, group, subname, snapname)
+	assert.NoError(t, err)
+	defer func() {
+		err := fsa.RemoveSubVolumeSnapshot(volume, group, subname, snapname)
+		assert.NoError(t, err)
+	}()
+
+	err = fsa.CloneSubVolumeSnapshot(
+		volume, group, subname, snapname, clonename,
+		&CloneOptions{TargetGroup: group})
+	assert.NoError(t, err)
+	defer func() {
+		err := fsa.RemoveSubVolume(volume, group, clonename)
+		assert.NoError(t, err)
+	}()
+
+	for done := false; !done; {
+		status, err := fsa.CloneStatus(volume, group, clonename)
+		assert.NoError(t, err)
+		assert.NotNil(t, status)
+		switch status.State {
+		case ClonePending:
+		case CloneInProgress:
+			assert.NotNil(t, status.ProgressReport.PercentageCloned)
+			assert.NotNil(t, status.ProgressReport.AmountCloned)
+			assert.NotNil(t, status.ProgressReport.FilesCloned)
```

Comment:
This seems OK, but I don't see anything asserting that the CloneInProgress state is reached. I think you could add an additional boolean (`wasInProgress`?) that gets asserted after the loop exits.

Reply:
Sure, I can add that.
Though, before submitting the PR (for the first time), I verified that the `InProgress` state is reached and we get the values parsed correctly. So, yes I have tested it locally.
But, for the sake of assurance I can update the test to make sure we are hitting the state _actually_